src/geocat/comp/climatology.py

```diff
@@ -284,39 +284,40 @@ def month_to_season(
             f"The {time_coord_name} axis length must be a multiple of {mod}.")
 
     seasons_pd = {
-        "DJF": ("QS-DEC", 1),
-        "JFM": ("QS-JAN", 2),
-        "FMA": ("QS-FEB", 3),
-        "MAM": ("QS-MAR", 4),
-        "AMJ": ("QS-APR", 5),
-        "MJJ": ("QS-MAY", 6),
-        "JJA": ("QS-JUN", 7),
-        "JAS": ("QS-JUL", 8),
-        "ASO": ("QS-AUG", 9),
-        "SON": ("QS-SEP", 10),
-        "OND": ("QS-OCT", 11),
-        "NDJ": ("QS-NOV", 12),
+        "DJF": ([12, 1, 2], 'QS-DEC'),
+        "JFM": ([1, 2, 3], 'QS-JAN'),
+        "FMA": ([2, 3, 4], 'QS-FEB'),
+        "MAM": ([3, 4, 5], 'QS-MAR'),
+        "AMJ": ([4, 5, 6], 'QS-APR'),
+        "MJJ": ([5, 6, 7], 'QS-MAY'),
+        "JJA": ([6, 7, 8], 'QS-JUN'),
+        "JAS": ([7, 8, 9], 'QS-JUL'),
+        "ASO": ([8, 9, 10], 'QS-AUG'),
+        "SON": ([9, 10, 11], 'QS-SEP'),
+        "OND": ([10, 11, 12], 'QS-OCT'),
+        "NDJ": ([11, 12, 1], 'QS-NOV'),
     }
     try:
-        (season_pd, season_sel) = seasons_pd[season]
+        ((month1, month2, month3), quarter) = seasons_pd[season]
     except KeyError:
         raise KeyError(
             f"contributed: month_to_season: bad season: SEASON = {season}. Valid seasons include: {list(seasons_pd.keys())}"
         )
 
-    start_date = dset[time_coord_name][0]
-    end_date = dset[time_coord_name][-1]
-
-    # Compute the three-month means, moving time labels ahead to the middle
-    # month.
-    month_offset = "MS"
-    dset_seasons = dset.resample({
-        time_coord_name: season_pd
-    },
-                                 loffset=month_offset).mean()
-
-    # Filter just the desired season, and trim to the desired time range.
-    compute_dset = dset_seasons.sel({
-        time_coord_name: dset_seasons[time_coord_name].dt.month == season_sel
-    }).sel({time_coord_name: slice(start_date, end_date)})
-    return compute_dset
+    # Filter data to only contain the months of interest
+    month_1st = dset.sel({time_coord_name: dset[time_coord_name].dt.month == month1})
```

Comment:
```suggestion
    data_filter = dset.sel({time_coord_name: dset[time_coord_name].dt.month.isin(months)})
```

where `months=(month1,month2,month3)`

Reply:
That is a much more concise way of doing this. I'm going to double check how it affects performance before accepting this suggestion.

Reply:
Thank you so much for this suggestion! I didn't realize that `.isin()` existed, and now the optimized version runs about twice as fast as the old one

Reply:
happy to help!